### PR TITLE
kernel: enable CONFIG_VFAT_FS 

### PR DESCRIPTION
**Pull request by aditjha-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

- Setting vfat module in kernel config to Y
- It was noticed that the kernel config was inherited from photon and other distros like ubuntu have vfat compiled in, and our aarch64 config does as well. Vfat is pretty widely used and will hopefully this change benefit more customers on top of the one that is already asking for it.

###### Change Log  

- Setting vfat module in kernel config to Y

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: 234931
- On VHD image of Mariner 2.0, `modprobe --first-time vfat` now shows an error because vfat is built into the kernel, prior to fix it would be able to load vfat module as it was available but not loaded.

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Could you expand on the rationale behind the change so we can understand why it was done in the future? ie, if we go to turn this back to a module in a year or two, it will help us remember why we decided to set it to a built-in instead of a module.

File: SPECS/kernel/config [hunk 1]
```diff
@@ -6517,9 +6517,9 @@ CONFIG_UDF_FS=y
 #
 # DOS/FAT/EXFAT/NT Filesystems
 #
-CONFIG_FAT_FS=m
+CONFIG_FAT_FS=y
 CONFIG_MSDOS_FS=m
-CONFIG_VFAT_FS=m
+CONFIG_VFAT_FS=y
```
Comment (aadhar-agarwal) [inline SPECS/kernel/config:6522]: What is the difference between `m` and `y` and what does CONFIG_FAT_FS mean exactly?

Comment (aditjha-msft) [reply SPECS/kernel/config:6522]: The difference between `m` and `y` is that the first makes sure the module is _available_ to be installed/uninstalled into the kernel, but the latter makes sure the module is _always_ installed into the kernel from boot up.

**Comment by aditjha-msft** [comment]
> Could you expand on the rationale behind the change so we can understand why it was done in the future? ie, if we go to turn this back to a module in a year or two, it will help us remember why we decided to set it to a built-in instead of a module.

It was noticed that the config was inherited from photon and other linux distros like ubuntu have it compiled in, as well as our aarch64 config. Also vfat is pretty widely used and will hopefully benefit more customers on top of the one that is already asking for it.

**Comment by dmcilvaney** [comment]
> 

Could you put something along those lines in the PR description? That is generally where people will look when trying to understand what was done.

**Review by christopherco — APPROVED** [review]

**Review by MaxBrodeurUrbas — APPROVED** [review]
🥇

**Pull request merged**